Microsoft.Dynamics365.UIAutomation.Api/Pages/FormJS.cs

```diff
@@ -0,0 +1,135 @@
+                null,
+                true,
+                typeof(NoSuchElementException), typeof(StaleElementReferenceException));
+        }
+
+        public bool SwitchToContent()
+        {
+            Browser.Driver.SwitchTo().DefaultContent();
+            //wait for the content panel to render
+            Browser.Driver.WaitUntilAvailable(By.XPath(Elements.Xpath[Reference.Frames.ContentPanel]));
+
+            //find the crmContentPanel and find out what the current content frame ID is - then navigate to the current content frame
+            var currentContentFrame = Browser.Driver.FindElement(By.XPath(Elements.Xpath[Reference.Frames.ContentPanel]))
+                .GetAttribute(Elements.ElementId[Reference.Frames.ContentFrameId]);
+
+            Browser.Driver.SwitchTo().Frame(currentContentFrame);
+
+            return true;
+        }
+
+        public BrowserCommandResult<T> GetAttributeValue<T>(string attributte)
+        {
+            var commandName = $"Get Attribute Value via Form JS: {attributte}";
+            string code = $"return Xrm.Page.getAttribute('{attributte}').getValue()";
+
+            return ExecuteJS<T>(commandName, code);
+        }
+
+        public bool SetAttributeValue<T>(string attributte, T value)
+        {
+            var commandName = $"Set Attribute Value via Form JS: {attributte}";
+            string code = $"return Xrm.Page.getAttribute('{attributte}').setValue(arguments[0])";
+
+            return ExecuteJS(commandName, code, value);
+        }
+
+        public bool Clear(string attribute)
+        {
+            var commandName = $"Clear Attribute via Form JS: {attribute}";
+            string code = $"return Xrm.Page.getAttribute('{attribute}').setValue(null)";
+
+            return ExecuteJS(commandName, code);
+        }
+
+        public BrowserCommandResult<Guid> GetEntityId()
+        {
+            var commandName = $"Get Entity Id via Form JS";
+            string code = $"return Xrm.Page.data.entity.getId()";
+
+            return ExecuteJS<string, Guid>(commandName, code, v =>
+            {
+                bool success = Guid.TryParse(v, out Guid result);
+                return success ? result : default(Guid);
+            });
+        }
+
+        public BrowserCommandResult<bool> IsControlVisible(string attributte)
+        {
+            var commandName = $"Get Control Visibility via Form JS: {attributte}";
+            string code = $"return Xrm.Page.getControl('{attributte}').getVisible()";
+
+            return ExecuteJS<bool>(commandName, code);
+        }
+
+        public BrowserCommandResult<bool> IsDirty(string attributte)
+        {
+            var commandName = $"Get Attribute IsDirty via Form JS: {attributte}";
+            string code = $"return Xrm.Page.getAttribute('{attributte}').getIsDirty()";
+
+            return ExecuteJS<bool>(commandName, code);
+        }
+
+        public enum RequiereLevel { Unknown = 0, None, Required, Recommended }
```

Comment:
Spelling mistake in the name of this enum?

Perhaps `RequiryLevel` or `RequiredLevel`?

Reply:
Hi  Matt. Thanks for the review. I already correct it in my branch